pgrx-pg-sys/src/submodules/transaction_id.rs

```diff
@@ -0,0 +1,74 @@
+//LICENSE Portions Copyright 2019-2021 ZomboDB, LLC.
+//LICENSE
+//LICENSE Portions Copyright 2021-2023 Technology Concepts & Design, Inc.
+//LICENSE
+//LICENSE Portions Copyright 2023-2023 PgCentral Foundation, Inc. <contact@pgcentral.org>
+//LICENSE
+//LICENSE All rights reserved.
+//LICENSE
+//LICENSE Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+use pgrx_sql_entity_graph::metadata::{
+    ArgumentError, Returns, ReturnsError, SqlMapping, SqlTranslatable,
+};
+
+pub type MultiXactId = TransactionId;
+
+/// An `xid` type from PostgreSQL
+#[repr(transparent)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(serde::Deserialize, serde::Serialize)]
+pub struct TransactionId(u32);
+
+impl TransactionId {
+    pub const INVALID: Self = Self::from_u32(0);
+    pub const BOOTSTRAP: Self = Self::from_u32(1);
+    pub const FROZEN: Self = Self::from_u32(2);
+    pub const FIRST_NORMAL: Self = Self::from_u32(3);
+    pub const MAX: Self = Self::from_u32(u32::MAX);
+
+    #[inline]
+    pub const fn from_u32(xid: u32) -> Self {
+        Self(xid)
+    }
+
+    #[inline]
+    pub const fn to_u32(self) -> u32 {
```

Comment:
Is it really fine? I would like to avoid any further name bikeshading as it's a little disaster even if another method added and this marked as deprecated. Shouldn't it be `into_u32`?

Reply:
honestly, we should call this `into_inner(self)` and then do the same over on `pg_sys::Oid` and close #2011.

And there's also no need for this `from_u32(xid: u32)` function either, because of `impl From<u32> for TransactionId` down below.

Reply:
`from_u32` is for `const` only since traits cannot have that modifier on the functions.  It can be removed, but then it won't be possible to use during compiler time.

If there's no need in `const fn`s then I see no reason in `from_smthg` and `into_smthg` at all as there are `from` and `into`. Makes sense?